crates/bevy_render/src/view/window.rs

```diff
@@ -166,7 +165,7 @@ pub struct WindowSurfaces {
 pub fn prepare_windows(
     // By accessing a NonSend resource, we tell the scheduler to put this system on the main thread,
     // which is necessary for some OS s
-    _marker: NonSend<NonSendMarker>,
```

Comment:
Why'd you change this instead of doing `init_non_send_resource`. It seems somewhat of a maintainability hazard to use an `Option` here since it seems reasonable that :sparkles: scheduler improvements :sparkles: could one day figure out that its `None` and ignore the `!Send`ness

Reply:
Mostly because it forcibly pollutes the `ComponentId` space with a marker resource. All this is doing is pinning the system to the main thread, so no reason to force it to allocate memory if we can just have it fail to fetch a `NonSend` of that type. Definitely a microoptimization. 

As for the scheduler optimizations, I don't see a way for us to detect that the resource is never inserted, especially when we have the presence of `ReflectResource`.

Reply:
Aight done.